projects/packages/autoloader/src/class-autoloader.php

```diff
@@ -79,7 +95,7 @@ function () use ( $plugins_handler, $cached_plugins ) {
 				// when a plugin is in the cache but not "active" when the autoloader loads.
 				// We also want to make sure that plugins which are deactivating are not
 				// considered "active" so that they will be removed from the cache now.
-				$active_plugins = $plugins_handler->get_active_plugins( false );
+				$active_plugins = $plugins_handler->get_active_plugins( false, ! $was_included_by_autoloader );
```

Comment:
Do we actually need to pass this variable here? There isn't really a case where we would want to register the current plugin as unknown during shutdown if it hadn't already been marked as such?

```suggestion
				$active_plugins = $plugins_handler->get_active_plugins( false, false );
```

Reply:
I think you're correct that there isn't a case where we would want to register the current plugin as unknown during shutdown.